src/inverter.rs

```diff
@@ -39,7 +40,42 @@ impl<'a> Inverter<'a> {
         }
     }
 
-    pub fn update_state(&mut self) -> Option<HMSStateResponse> {
+    pub fn mock_state(&mut self) -> Option<HMSStateResponse> {
```

Comment:
The test code should be separated from the implementation. This could be done in a tool simulating the behavior of an actual inverter. Issue #11 is tracking such a feature. This should be dropped from the PR.